docs: add docs on env variables for serverless (#149)
tophat/sanity-runner: service/README.md

# sanity-runner (service)

The sanity-runner service is a lambda function that sits in AWS. It takes a [jest-puppeteer](https://github.com/smooth-code/jest-puppeteer) test file as input and returns the result of the test.

# Configuration

> We currently use [serverless](https://serverless.com/framework/docs/) to deploy to AWS lambda. We use Environment variables to help define resource names allowing for easy deployment of multiple sanity-runners across the same envrionment.

Reviewer: ```suggestion
We currently use [serverless](https://serverless.com/framework/docs/) to deploy to AWS lambda. We use environment variables to help define resource names allowing for easy deployment of multiple sanity-runners across the same environment.
```

| `SERVERLESS_FUNC`           | sanity-runner-${self:custom.stage}-${self:custom.tag} | Name of the Lambda function created |
| `SERVERLESS_STAGE`          | self:provider.stage                                   | Defines serverless stage |
| `SERVERLESS_VPC_ENABLED`    | "false"                                               | If "true" will deploy the lambda to a VPC. Requires `SG_ID` and `SUBNET_ID` to work |
| `SERVERLESS_SG_ID`          | "not_set"                                             | Security Group to associate with the lambda function. Requires `VPC_ENABLED="true"` |
| `SERVERLESS_SUBNET_ID`      | "not_set                                              | Subnet to associate with the lambda function. Requires `VPC_ENABLED="true"` |

## Quick Start

If wanting to just quickly deploy the lambda without conflicting with anything, just source [serverless.env](./serverless.env). This will set `SERVERLESS_TAG` and `SERVERLESS_STAGE` based on AWS account + git branch name.

# Deployment

## Install Depdendancies

```
make install
```

> If Running on a non-linux system, we reccomend using `make install-mac` instead of make install. This will compile the dependancies in a linux docker container, ensuring they are compiled for the correct OS. (other wise they will compile for Mac OS and break when published to AWS Lambda)

Reviewer: ```suggestion
If Running on a non-linux system, we recommend using `make install-mac` instead of make install. This will compile the dependencies in a linux docker container, ensuring they are compiled for the correct OS. (other wise they will compile for Mac OS and break when published to AWS Lambda)
```

> If Running on a non-linux system, we recommend using `make install-mac` instead of make install. This will compile the dependencies in a linux docker container, ensuring they are compiled for the correct OS. (other wise they will compile for Mac OS and break when published to AWS Lambda)

Reviewer: ```suggestion
If running on a non-linux system, we recommend using `make install-mac` instead of make install. This will compile the dependencies in a linux docker container, ensuring they are compiled for the correct OS. (other wise they will compile for Mac OS and break when published to AWS Lambda)
```
